### ISSUE TITLE
@html does not update on hydration

### ISSUE THREAD

**Issue by hobbes7878** [reuters-graphics/graphics-components#148]
Inventory of the components using `@html` tags that will need to be updated to get around this issue:
- [x] BodyText
- [x] EndNotes
- [x] GraphicBlock
- [x] Headline
- [x] InfoBox
- [x] PhotoPack
- [x] Scroller
- [x] SimpleTimeline
- [x] Table

---

There's a few longstanding issues on the Svelte repo about this, amazingly.

A workaround is to set `innerHtml` with an action, which is what I guess we'll have to do before we can have live updating previews from RNGS.io working.. 

```svelte
<script>
function setInnerHTML(node, html) {
    node.innerHTML = html;
    return {
      destroy() {
        node.innerHTML = '';
      },
    };
  }
</script>

<div use:setInnerHTML="{marked.parse(text)}"></div>

<style>
  div {
    display: contents;
  }
</style>
```

Honestly, amazed this hasn't come up for me before...

Will likely need to scrub all our components for this, but for sure BodyText is first up.

**Comment by hobbes7878** [comment]
This is tricky b/c this text is not prerendered...

Can do this to get it back, but I need to test if this is gonna lead to flashes on prod stuff.

```svelte
<script>
import { building } from '$app/environment';

function setInnerHTML(node, html) {
    node.innerHTML = html;
    return {
      destroy() {
        node.innerHTML = '';
      },
    };
  }
</script>

{#if building}
  <div>
    {@html marked.parse(text)}
  </div>
{:else}
  <div use:setInnerHTML="{marked.parse(text)}"></div>
{/if}

<style>
  div {
    display: contents;
  }
</style>
```

**Comment by hobbes7878** [comment]
OK, it looks like the above workaround doesn't lead to flashes and still prerenders. Idea would be to make a standalone component for markdown text, I suppose, and replace all the `@html` refs with it.

Trouble is `building` state can't rely on SvelteKit variables, which causes trouble in SREP land. Possible solutions, all bad:

 - Make a store that we set in SvelteKit projects to mirror `building`..
   -  There is no equivalent in SREP Vite-only projects, so we'd need to set a default that would always be true for SREPs. If that's using `@html` then we've locked SREPs out of live endpoint updates.
 - Actively check for Node v. Browser context by checking `window` or something??
   - Really hacky...
 - Look at alternative solutions that actively parse markdown into Svelte text components like [svelte-markdown](https://www.npmjs.com/package/svelte-markdown)
   -  ... but that's a monstrous dependency, not super actively maintained.

Sucks, all of this.
